build/Tests.lua

```diff
@@ -104,27 +104,37 @@ function SetupMono()
   end
 
   includedirs { path.join(monoDir, "include", "mono-2.0") }
-  libdirs { path.join(monoDir, "lib") }
-  links { "monosgen-2.0" }
-
+  monoLibdir = path.join(monoDir, "lib")
+  libdirs { monoLibdir }
+  if(os.is("windows")) then
+    libFiles = os.matchfiles(path.join(monoLibdir, "monosgen-2.0.lib"))
+    table.insert(libFiles, os.matchfiles(path.join(monoLibdir, "mono-2.0-sgen.lib")))
+    links { libFiles }
+  else
+    links { "monosgen-2.0" }
+  end
+  
   filter { "system:macosx" }
     links { "CoreFoundation.framework" }
 
   filter {}
 end
 
 function SetupTestProjectC(name, depends)
-  if string.starts(action, "vs") and not os.is("windows") then
-    return
-  end
+  -- if string.starts(action, "vs") and not os.is("windows") then
+    -- return
+  -- end
 
   project(name .. ".C")
-    location "mk"
 
     kind "SharedLib"
     language "C"
 
+    if string.starts(action, "vs") then
+    defines { "MONO_EMBEDDINATOR_DLL_EXPORT", "MONO_DLL_IMPORT", "MONO_M2N_DLL_EXPORT" }
```

Comment:
No need for the if here, just rename `MONO_M2N_DLL_EXPORT` to `MONO_EMBEDDINATOR_DLL_EXPORT`.